numpy/testing/_private/utils.py

```diff
@@ -821,7 +821,12 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
 
                     # note: this definition of relative error matches that one
                     # used by assert_allclose (found in np.isclose)
-                    max_rel_error = (error / abs(y)).max()
+                    # Filter values where the divisor would be zero
+                    nonzero = y != 0
+                    if all(~nonzero):
+                        max_rel_error = array(inf)
```

Comment:
OK, I guess this seems right if `y` happens to be all zero. I believe the result here should be just `inf` and not an array with `inf` inside, `max` also returns a scalar. (Although this is probably just printed, so it may not even make a difference.

Reply:
Could you add a test for this branch (assuming it does not exist)?

Reply:
```max_rel_error``` is passed to ```array2string``` and then printed, that's why I made it an array (although I guess ```float_(inf)``` would also work, but it makes no difference)
For the ```inf``` branch there is already a test: ```testing.tests.test_utils.TestAlmostEqual.test_error_message``` which tests this exact scenario of y being all zeros.
I've also written a test for the original issue ```TestAssertAllclose.test_report_max_relative_error```